### ISSUE TITLE
Keyboard does not work

### ISSUE THREAD

**Issue by shantigilbert** [uobikiemukot/yaft#41]
I am trying to implement this in my image, I run the terminal from Emulationstation, and it displays fine, but they keyboard does not work :( 

My keyboard works fine with other programs, its just YAFT.

If I run it from SSH the keyboard works, altho I get this message

```
WARN<< ioctl: VT_SETMODE failed (maybe here is not console)
WARN<< ioctl: KDSETMODE failed (maybe here is not console)
```
But I can use it and it kinda works, but if I try to run any program in the terminal I get an error that there was a problem opening the terminal "yaft-256colors" I am hoping this is because it is running from SSH.

Is there anything I need to setup to make YAFT recognize my keyboard? maybe an ENV ?

**Comment by uobikiemukot** [comment]
Hi,

Sorry I don't know what Emulationstation is.

Anyway, yaft cannot work via SSH. because framebuffer is local resource. it's not available from remote machine. If you want to use yaft, you need to login the machine locally (without using SSH)

> I get an error that there was a problem opening the terminal "yaft-256colors"

To fix this problem, you should install terminfo like this: `# tic info/yaft.src`

**Comment by shantigilbert** [comment]
Thanks, I know that YAFT does not work from ssh (but I do see it running on my TV) but I was making the point that the keyboard works fine if I run it like that, but the keyboard it does not if I run it locally

>>To fix this problem, you should install terminfo like this: # tic info/yaft.src

I will try this! will this fix the keyboard issue as well?

**Comment by uobikiemukot** [comment]
> Thanks, I know that YAFT does not work from ssh (but I do see it running on my TV) but I was making the point that the keyboard works fine if I run it like that, but the keyboard it does not if I run it locally

Maybe you are using yaft via SSH, but your TV is connected to the machine (yaft running on) locally. it's very rare situation. I have never used yaft in that situation.

>> To fix this problem, you should install terminfo like this: # tic info/yaft.src
>
> I will try this! will this fix the keyboard issue as well?

terminfo will fix specific input problem. Some movements (cursor movement, backspace, ctrl-H, etc...) work correctly with valid terminfo. And some text based software uses terminfo for efficient rendering.

**Comment by shantigilbert** [comment]
Yes if I start YAFT from SSH on my image (I use https://github.com/shantigilbert/CoreELEC) I can see it on my TV which is connected to my Amlogic box using HDMI, keyboard and everything seems to work fine. 

But that is not how I want to use it, I want to run it locally (run from a script) I can see the terminal correctly on the screen, but the keyboard is not working :(

I will try your suggestion and get back at you ASAP.

Thanks!

**Comment by shantigilbert** [comment]
That solution made programs work, even if I run it from SSH.

But the keyboard does not work if I run it locally, it only works if I run it from SSH. any other idea on how to fix? 

I think my issue is related to this issue as well https://github.com/uobikiemukot/yaft/issues/15 but I have no idea on how to implement your fix, do I need to change ALL references from STDIN_FILENO to "/dev/input/event6" or just the one you mentioned? I also have no idea how to do the character conversion you mentioned :/

**Comment by uobikiemukot** [comment]
I can fix yaft to support a different input source, if  there is a way to run yaft on CoreElec by myself. Maybe I can test CoreElec by using RaspberryPi? (if so, maybe it take a while)

> I think my issue is related to this issue as well #15 but I have no idea on how to implement your fix, do I need to change ALL references from STDIN_FILENO to "/dev/input/event6" or just the one you mentioned? I also have no idea how to do the character conversion you mentioned :/

I also don't know how to fix this problem at the moment :crying_cat_face:

**Comment by shantigilbert** [comment]
You would need to run LibreELEC (https://github.com/LibreELEC/LibreELEC.tv) if you want to run it on a RaspberryPi  which is basically the same thing, but you would need to either use autostart.sh to start YAFT before Kodi, or make a sh file and use an addon to run it. The one I am working on does not have Kodi (special build) which makes Yaft a lot more useful. 

I will keep investigating and let you know if I come up with something that works. Thanks!

**Comment by shantigilbert** [comment]
@uobikiemukot did you ever found a solution for this?
